src/ghostfat.c

```diff
@@ -312,29 +331,44 @@ void uf2_read_block (uint32_t block_no, uint8_t *data)
       d->updateTime = __DOSTIME__;
       d->updateDate = __DOSDATE__;
       d->startCluster = startCluster & 0xFFFF;
-      d->size = (inf->content ? strlen(inf->content) : UF2_SIZE);
+      d->size = (inf->content ? strlen(inf->content) : UF2_BYTE_COUNT);
     }
   }
   else if ( block_no < BPB_TOTAL_SECTORS )
   {
-    sectionIdx -= FS_START_CLUSTERS_SECTOR;
-    if ( sectionIdx < (NUM_FILES - 1) * BPB_SECTORS_PER_CLUSTER )
+    sectionRelativeSector -= FS_START_CLUSTERS_SECTOR;
+    uint32_t sectionRelativeClusterIndex = sectionRelativeSector / BPB_SECTORS_PER_CLUSTER;
+
+    if ( sectionRelativeClusterIndex < (NUM_FILES - 1) )
     {
-      memcpy(data, info[sectionIdx / BPB_SECTORS_PER_CLUSTER].content, strlen(info[sectionIdx / BPB_SECTORS_PER_CLUSTER].content));
+      // WARNING -- code presumes first data cluster == first file, second data cluster == second file, etc.
+      struct TextFile const * inf = &info[ sectionRelativeSector / BPB_SECTORS_PER_CLUSTER ];
+      size_t fileContentStartOffset = (sectionRelativeSector % BPB_SECTORS_PER_CLUSTER) * BPB_SECTOR_SIZE;
+      size_t fileContentLength = strlen(inf->content);
+
+      if (fileContentLength > fileContentStartOffset) {
+        size_t bytesToCopy = fileContentLength - fileContentStartOffset;
+        const char * dataStart = (inf->content) + fileContentStartOffset;
+        STATIC_ASSERT(sizeof(char) == 1); // technically not required to be true
```

Comment:
this should always be true for embedded, though i don't really needs it. We could just declare dataStart as either `const void*` or `const uint8_t*`

Reply:
OK, I will update to one of these, likely `const void *` unless you have other preference.

Reply:
I would use `const void*` as well.